### ISSUE TITLE
P0 architecture: point-in-time news, social, filings NLP, manipulation defenses and alternative data

### ISSUE THREAD

**Issue by leon36000** [leon36000/MarketOS-#45]
# R12 preclosure architecture finding

R12 refutes a generic news/social adapter, mutable “latest content,” universal sentiment scores, silent event deduplication, model-name-only inference receipts, bot labels treated as ground truth, public access treated as unrestricted rights, reconnect treated as recovery and alternative data without consent/coverage lineage.

This issue records the target architecture and permanent adversarial corpus. It does **not** authorize provider procurement, production endpoints, credentials, model adoption or a new sibling implementation branch.

## Required evidence chain

```text
immutable raw content bytes
→ platform/provider/native identity and cursor receipt
→ append-only lifecycle events
→ point-in-time availability and knowledge state
→ claim/narrative and event-cluster evidence
→ content-addressed NLP inference receipt
→ calibrated coordination or alternative-data evidence
→ rights/deletion receipts
→ content-addressed point-in-time dataset
```

## P0 invariants

1. Persist exact raw payload bytes before parsing, enrichment or inference.
2. A URL, accession or API endpoint is a locator, not source-byte authority.
3. Story, post, filing, media, account/channel and provider identities remain typed and source-native.
4. Preserve native IDs, version IDs, cursors, partitions, sequence numbers and resume tokens.
5. Create, edit, delete, tombstone, correction and retraction are append-only lifecycle events.
6. An edit or delete never silently overwrites a historically observed version.
7. Source-event, publish, edit, delete, capture, first-verified-availability, knowledge, valid, system and dataset-cutoff times remain distinct.
8. Current platform state cannot leak into an earlier historical cutoff.
9. Thread, reply, quote, repost, section, table, footnote, layout and media context remain evidence.
10. Claims bind speaker/source, evidence span, target entity/instrument, support, contradiction, qualification and retraction edges.
11. Cross-provider event clustering is an evidence graph decision; source disagreement remains explicit.
12. Overmerge and undermerge are permanent adversarial tests.
13. A universal sentiment score is rejected. Target, aspect, stance, polarity, uncertainty, emotion, quotation scope, language and horizon remain explicit.
14. Every NLP inference binds model, tokenizer, prompt, code, configuration, input-artifact and output SHA-256.
15. Confidence requires calibration, abstention, alternative labels and review state.
16. Bot or coordinated-behavior classifications are probabilistic hypotheses, not ground truth.
17. Coordination evidence preserves temporal, content, URL-sharing, account, network, cross-platform, financial-incentive and counterevidence edges.
18. Alternative-data observations preserve collection method, consent/lawful basis, spatial/temporal coverage, sampling, drift, privacy, rights and deletion lineage.
19. Display, non-display, redistribution, internal/external derived data, historical storage, cloud processing, development, AI/ML training and audit export are distinct rights classes.
20. Unknown, missing or expired rights fail closed.
21. Deletion and withdrawal propagate to raw access, indexes, caches, embeddings, features, model-training sets, datasets and exports through append-only receipts.
22. Transport reconnect, subscription recovery, cursor recovery, replay and state convergence are separate proofs.
23. Backpressure, gap age, replay lag, source-to-capture latency, queue depth and p50/p95/p99/p999 tails are explicit.
24. Every dataset binds source-artifact hashes, lifecycle-event hashes, knowledge cutoff, rights, deletion policy, model receipts, cluster receipts, code/config hashes and dataset SHA-256.
25. Own-order truth preserves four external channels:

```text
ORDER_ENTRY
+ MARKET_DATA
+ DROP_COPY_OR_TRADE_CAPTURE
+ CLEARING_OR_POST_TRADE
```

26. The local journal remains additional internal evidence only.
27. No vendor, platform, stream processor, store, graph engine or model runtime is selected before a common MarketOS bakeoff.
28. No production route exists while `live_trading=HARD_LOCKED`.

## Permanent adversarial corpus

The R12 corpus includes at least:

- raw payload loss and source spoofing;
- native-ID reuse and conflicting duplicates;
- sequence gaps, cursor gaps and false reconnect recovery;
- edit overwrite, ignored deletion and lost retraction;
- publish/edit/delete/capture/knowledge-time inversion;
- current-state leakage into historical datasets;
- thread, quotation, filing-section and media context loss;
- entity collisions and claim misattribution;
- sarcasm, negation, multilingual and language-drift failures;
- model/tokenizer/prompt/config drift;
- uncalibrated confidence and suppressed abstention;
- bot false positives and coordination false negatives;
- event-cluster overmerge and undermerge;
- platform terms misread as redistribution or AI-training rights;
- deletion remaining in embeddings, features or model datasets;
- alternative-data consent, coverage and selection bias;
- backpressure drops and recovery divergence;
- deterministic dataset rebuild mismatch;
- runtime-local artifact receipts falsely presented as global bytes.

## Local preclosure corpus status

```yaml
classification: VERIFIED_LOCAL_PRECLOSURE_RESEARCH_BYTES
phase_event_appended: false
minimum_contracts:
  registered_source_locators: ">=240"
  requirements: ">=220"
  failure_modes: ">=300"
  permanent_tests: ">=360"
source_byte_authority: 0
technology_adoptions: 0
implementation: NOT_STARTED
```

The local source registry deliberately distinguishes locators from retrieved, hashed source documents. Exact pack, manifest, receipt and verification hashes are retained in the local machine-readable closure receipt and will be promoted only after fresh memory binding.

## Technology status

```yaml
generic_news_social_adapter: REJECT
mutable_latest_content: REJECT
universal_sentiment_score: REJECT
silent_event_deduplication: REJECT
model_name_only_receipt: REJECT
bot_label_as_ground_truth: REJECT
public_access_implies_rights: REJECT
reconnect_equals_recovery: REJECT
selected_vendor: NONE
selected_platform: NONE
selected_stream_processor: NONE
selected_store: NONE
selected_graph_engine: NONE
selected_model_runtime: NONE
technology_adoptions: 0
production_connectivity: REJECT_HARD_LOCK
```

## Integration boundary

Do not create a fourth sibling implementation branch. Preserve:

```text
repair integration manifest
→ PR29
→ replay PR33
→ replay PR30
→ reconcile PR21/main
```

After the integration authority is rebuilt, implementation may be decomposed into bounded TDD slices with non-overlapping ownership.

## Immutable boundaries

```yaml
live_trading: HARD_LOCKED
profitability: UNPROVEN
production_ready: false
project_complete: false
false_done: FORBIDDEN
stubs: FORBIDDEN
```

**Comment by leon36000** [comment]
## R12 local preclosure corpus verified — no phase promotion yet

A fresh current-runtime corpus now exists for news feeds, social networks, filings NLP, narrative/sentiment, event clustering, manipulation defenses, alternative data, point-in-time knowledge, rights/deletion and low-latency ingestion.

```yaml
classification: VERIFIED_LOCAL_PRECLOSURE_RESEARCH_BYTES
phase_event_appended: false
implementation: NOT_STARTED
technology_adoptions: 0
source_validator: PASS
contract_tests: PASS
extracted_validator: PASS
extracted_contract_tests: PASS
zip_integrity: PASS
deterministic_rebuild: PASS_BYTE_IDENTICAL_TWICE
```

The TDD suite was authored first. A clean RED fixture failed because the corpus was absent. The generator was then completed without reducing thresholds; the first GREEN corpus passes at least 240 classified locators, 220 requirements, 300 failure modes, 360 permanent tests, 60 findings, 60 technology decisions, 28 coverage gaps, 30 negative searches, 30 vendor candidates and 35 software candidates.

The source registry deliberately records locators rather than pretending they are retrieved source documents:

```yaml
retrieved_byte_verified_sources: 0
source_locator_is_byte_authority: false
unknown_rights: DENY_FAIL_CLOSED
```

Core contracts now include immutable raw content, append-only create/edit/delete/tombstone/correction/retraction events, separate publish/capture/availability/knowledge times, claim and narrative graphs, content-addressed inference receipts, calibrated coordination hypotheses, alternative-data consent/coverage/drift lineage, deletion propagation and explicit cursor/gap recovery.

Exact pack, manifest, receipt, log and sidecar hashes are stored in the machine-readable local closure receipt and the preceding exact-receipt comment. No R12 phase event will be appended until those bytes are freshly bound to Neon/Mongo and a causal R13 bootstrap snapshot, audits and checkpoint are produced.

```yaml
live_trading: HARD_LOCKED
profitability: UNPROVEN
project_complete: false
production_ready: false
```

**Comment by leon36000** [comment]
## R12 generation 3 — current-runtime bytes rebuilt; promotion failed closed

The current sandbox had lost the previous R12 work directory and deliverables. No remembered hash was reused. The complete corpus was rebuilt from the TDD contracts in the current runtime, sidecars were regenerated, extraction was repeated in a clean directory, and two deterministic rebuilds were byte-identical.

```yaml
rebuild_generation: 3
classification: VERIFIED_LOCAL_PRECLOSURE_RESEARCH_BYTES
current_runtime_files_present: true
phase_event_appended: false
R13_snapshot_created: false
R12_audits_created: false
R12_transition_checkpoint_created: false
```

The exact byte counts and SHA-256 values are authoritative only in the current machine-readable files:

```text
MarketOS_Relaunch_12_Closure_v1_Receipt.json
MarketOS_R12_Closure_v1_Hashes.json
MarketOS_R12_Local_Preclosure_Status.json
```

A direct Neon promotion was attempted but the sandbox had no usable direct Neon CLI/credential path. The operation failed before any write. Connector readback then proved that the database still had:

```yaml
current_phase: R11.3
current_phase_event_sha256: 8e758aedaf807e4c6eeb52edea08ff0253ba97be1e406158ac34f8afe2bfc563
R12_phase_events: 0
R12_decisions: 0
R12_open_loops: 0
R12_artifact_receipts: 0
R12_markdown_documents: 0
promotion_result: FAILED_CLOSED_NO_PARTIAL_WRITES
```

A preclosure source, checkpoint and Mongo qualification receipt now preserve the rebuild generation, present-file status, exact-hash authority filenames, verification results and unresolved atomic binding actions:

```yaml
neon_preclosure_checkpoint: 12121212-1212-4212-8212-121212121200
mongo_receipt: marketos-r12-preclosure-local-bytes
```

The next valid transition remains:

```text
load exact values from current receipt bytes
→ append R12 phase + Markdown atomically
→ append decisions/open loops/artifact receipts
→ exact Mongo mirror
→ causal R13 pre-audit snapshot
→ recall/parity/artifact/full audits
→ post-audit snapshot
→ transition checkpoint
```

No R12 closure or implementation claim is made.

```yaml
live_trading: HARD_LOCKED
profitability: UNPROVEN
false_done: FORBIDDEN
stubs: FORBIDDEN
project_complete: false
production_ready: false
```

**Comment by leon36000** [comment]
## Mobile Sol R12/42 final current-runtime qualification — research closed, R13 bootstrap activated

R12 is now closed at the research-contract and current-runtime memory-qualification levels. This issue remains open because implementation, provider rights, ground truth, bakeoffs and global portable convergence are incomplete.

### Semantic authority

```yaml
revision_1:
  event_sha256: 632dc2ec06e68c00bb23dd374fcb2d2f9658c25a3abc41ef95bc04aaaebdf6a5
  pack_sha256: a370c378be1bbaaf7a229c9c8cee5d0a2273791bae92ca1209ac220301531aa8
  authority: REMOTE_RECEIPT
revision_2:
  event_sha256: 15ed23bf7c0238691c00eb210674b3870c0178d653986b2dcb28d194fcc18587
  pack_sha256: 8146b8e698b9537c028ffd4c25222b2b405607d6e0f991fb81095b797e7920ba
  transition_pack_sha256: 034356b853c41375a89869293847a72e7c4a002f077b9504c8b1a5f59c6d9f13
  authority: CURRENT_RUNTIME_BYTE_VERIFIED
revision_3:
  event_sequence: 37
  event_sha256: 08ff66fb5f630a654d624f4b85927834fbbdeeb71b5c17891d3691513cc181a9
  pack_sha256: 372e7dfa313002dde8525c06370091e2b4edc2094ae36f99d4a1be2d24b9debd
  transition_pack_sha256: 013988931a4f83e13f8c81908fd0e2b8e75e54b47dd118520ac1539015ac6eec
  authority: REMOTE_RECEIPT
research_semantics: CLOSED_THREE_COMPATIBLE_LINEAGES
count_rule: ARTIFACT_SCOPED_NOT_SUMMED_NOT_MAXIMIZED
global_single_portable_authority: false
```

### Runtime-local audit authority

Audits 63–66 are preserved but do not qualify this runtime: their historical recall used noncurrent R02/M01 heads and their artifact audit qualified `372e…/0139…` bytes absent here.

```yaml
audit_authority_decision: 4a1f82c2acad55a9c3dbe155548228de59fcadd1464842891762ddb0935e341d
repair_loop_resolution: 3541e755648afad9a20a20a25dca303225916b122ba6d633c1552a4df537a8d2
67_HISTORICAL_RECALL: 61bcacbcc9afcaf9bf33ef0ff198caa8b0e91f46b70083b0ac919e3ac28fd7af
68_CROSS_STORE_PARITY: 934ad9bf62f213154ba2f7bacd84a280b30e050cd78bc5dc271d54ce9b3edf3d
69_ARTIFACT_HASH: 2844fb3e2e47f55413d7dd31ec04782d41cfb2fe53bbd5ecd8d9c44f756abc1b
70_FULL_MEMORY_AUDIT: 9ba7251386f8258729b01ccde48b6a36cab7b299b0975231b95b05e6fda9d56c
71_FINAL_CROSS_STORE_PARITY: 43be06cbf55ee43ca0342f9fb4adde97cb43c8ac73c1fb12243eda7d4e8b887e
audit_index_sha256: a54695d22447f33faa066eb5f3b19b395178374c28fcb1eeea44c55f30b853d5
```

### Exact cross-store repair

The final audit found five Neon audit identities absent or malformed in Mongo. Audits 25, 26, 27, 50 and 65 were restored by exact Neon UUID/SHA. A corrupted 65-character SHA document was retained but excluded rather than overwritten.

```yaml
phase_events: 31/31
decision_events: 231/231
active_decisions: 161/161
open_loop_events: 166/166
active_open_loops: 62/62
artifact_receipts: 270/270
markdown_documents: 38/38
Neon_audit_hashes: 69/69
bootstrap_snapshots: 34/34
logical_key_uniqueness: PASS
```

### Qualified R13 bootstrap

```yaml
snapshot_id: 1ed9c62f-c68f-4a2d-9c1c-c056703a821d
snapshot_sequence: 34
snapshot_sha256: e241796dbfc59ef9d7e0f451bb7157fef6c7829900742260af5e00071fa264d9
memory_manifest_sha256: a54695d22447f33faa066eb5f3b19b395178374c28fcb1eeea44c55f30b853d5
captured_at: 2026-08-24T23:48:38.681533Z
recorded_at: 2026-08-24T23:48:38.688786Z
checkpoint: 12121212-1212-4212-8212-121212121234
checkpoint_created_at: 2026-08-24T23:53:59.765Z
causal_order: PASS
```

### Current-runtime bytes

```yaml
research_pack_sha256: 8146b8e698b9537c028ffd4c25222b2b405607d6e0f991fb81095b797e7920ba
research_pack_bytes: 70965
transition_pack_sha256: 034356b853c41375a89869293847a72e7c4a002f077b9504c8b1a5f59c6d9f13
transition_pack_bytes: 80196
fresh_zip_json_sidecar_verification: PASS
registered_source_locators: 289
retrieved_byte_verified_sources: 0
technology_adoptions: 0
```

No provider, platform, stream processor, store, graph engine or model runtime is selected.

```yaml
R13_phase_events: 0
R13_status: PREPHASE_READY
implementation: NOT_STARTED
capacity: UNPROVEN
live_trading: HARD_LOCKED
profitability: UNPROVEN
project_complete: false
production_ready: false
```

**Comment by leon36000** [comment]
## R12 generation 3 — atomic promotion bundle ready, not executed

The current-runtime preclosure bytes have been converted into a deterministic, self-validating promotion package. This is an execution bundle, **not** an execution receipt and not a closure claim.

```yaml
bundle: MarketOS_R12_Atomic_Promotion_Generation3_2026-08-24.zip
classification: VERIFIED_EXECUTION_BUNDLE_NOT_EXECUTION_RECEIPT
validator: PASS
deterministic_rebuild: PASS_BYTE_IDENTICAL
phase_event_appended: false
R12_closed: false
```

The package contains:

```text
R12_ATOMIC_PROMOTION_MANIFEST.json
01_NEON_R12_ATOMIC_PROMOTION_STAGE_A.sql
02_MONGO_AND_CLOSURE_PROTOCOL.json
validate_promotion_bundle.py
BUNDLE_HASHES.json
VALIDATION.log
README.md
```

Stage A is one fail-closed Neon transaction with:

- exact bootstrap assertions against R11.3, snapshot 32 and checkpoint `11111111-1111-4111-8111-111111111132`;
- the R12 Markdown authority document;
- one R12 phase event;
- 18 binding decisions;
- 10 explicit open loops;
- exact current-runtime artifact receipts generated from the present files;
- post-write count and hash assertions;
- rollback on any mismatch.

The closure protocol forbids a closure claim until exact Neon rows are mirrored to Mongo, logical-key uniqueness and critical hashes match, an R13 causal pre-audit snapshot is created, current-runtime audits pass, a post-audit snapshot is sealed and the transition checkpoint follows it.

A separate `MarketOS_R12_Direct_Execution_Attempt.json` records whether any direct client path executed the SQL. Database readback remains the only authority; no execution claim is inferred from the presence of the bundle.

```yaml
live_trading: HARD_LOCKED
profitability: UNPROVEN
false_done: FORBIDDEN
stubs: FORBIDDEN
project_complete: false
production_ready: false
```
